modules/search-pipeline-common/src/yamlRestTest/resources/rest-api-spec/test/search_pipeline/30_filter_query.yml

```diff
@@ -81,3 +81,23 @@ teardown:
         }
   - match: { hits.total.value: 1 }
   - match: { hits.hits.0._id: "2" }
+
+  # Should work with inline query
+  - do:
```

Comment:
Isn't this test run in the mixed cluster test where it might hit a server running 2.8? I don't see any failures, but the behavior is non-deterministic depending on how a request is routed at runtime during the test.

Reply:
Maybe that only applies to the tests under the rest-api-spec root?

I had to guard the test that I added there behind a [version check](https://github.com/opensearch-project/OpenSearch/blob/main/rest-api-spec/src/main/resources/rest-api-spec/test/search_pipeline/10_basic.yml#L4), because it was failing in the mixed cluster tests, but haven't added any version checks for these module tests.

(Actually that test should be updated to only skip versions less than 2.7 now.)